### PR TITLE
Reworking avro dimension loader to load schema from avro file

### PR DESCRIPTION
**Pull request by will-lauer**
Removing separate schema file and using base schema from the avro file instead.

### CODE REVIEW

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java [hunk 1]
```diff
@@ -104,34 +90,24 @@ private boolean doesSchemaContainAllDimensionFields(Dimension dimension, String
...
-
-            DataFileReader<GenericRecord> dataFileReader;
-            String filePath = avroSchemaPath;
-            try {
-                // Creates an AVRO schema object based on the parsed AVRO schema file (.avsc)
-                Schema schema = new Schema.Parser().parse(new File(filePath));
-
-                // Creates an AVRO DatumReader object based on the AVRO schema object
-                DatumReader<GenericRecord> datumReader = new GenericDatumReader<>(schema);
+        // Creates an AVRO DatumReader object based on the AVRO schema object
```
Comment (cdeszaq) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java]: Comment not correct, as there's no relevant schema here...

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java [hunk 2]
```diff
@@ -69,29 +62,22 @@ public AvroDimensionRowParser(DimensionFieldNameMapper dimensionFieldNameMapper,
      * </code></pre>
```
Comment (will-lauer) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java]: The JSON listed is still how the avro schema is represented internally in the avro file, so it is still useful to have as a comment.

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java [hunk 3]
```diff
@@ -69,29 +62,22 @@ public AvroDimensionRowParser(DimensionFieldNameMapper dimensionFieldNameMapper,
...
      *
      * @param dimension The dimension object used to configure the dimension
-     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     * @param avroSchema The path of the AVRO schema file (.avsc)
```
Comment (will-lauer) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java]: I'll fix it.

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java [hunk 4]
```diff
@@ -69,29 +62,22 @@ public AvroDimensionRowParser(DimensionFieldNameMapper dimensionFieldNameMapper,
...
+        // Get main record from schema
+        List<Schema.Field> fields = avroSchema.getFields();
 
-        // Populating the set of avro field names
-        Set<String> avroFields = StreamSupport.stream(jsonAvroSchema.spliterator(), false)
-                .map(jsonNode -> jsonNode.get("name"))
-                .map(JsonNode::asText)
-                .collect(Collectors.toSet());
+        // Extract field names
+        Set<String> avroFields = StreamSupport.stream(fields.spliterator(), false)
```
Comment (will-lauer) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java]: fixing

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java [hunk 5]
```diff
@@ -69,29 +62,22 @@ public AvroDimensionRowParser(DimensionFieldNameMapper dimensionFieldNameMapper,
...
+                        .map(field -> field.name())
```
Comment (will-lauer) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java]: fixing

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java (hunk 1 above)
Comment (will-lauer) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java]: fixing

**Comment by michael-mclawhorn** [comment]
👍

**Pull request merged**